src/blop/objectives.py

```diff
@@ -162,25 +202,23 @@ def constrain(self, y: torch.Tensor) -> torch.Tensor:
         else:
             return torch.tensor([value in self.constraint for value in np.atleast_1d(y)])
 
+    @property
+    def all_valid(self) -> bool:
+        return not (self.validity_conjugate_model and self.validity_probability)
```

Comment:
Double checking intention on logic here, maybe worth adding it as a comment with explanation.

If either the `validity_conjugate_model` or the `validity_probability` are not defined, then all are valid?

Reply:
The `Objective` gets its model from the `_construct_model()` method in the `Agent`, which gives it a `validity_conjugate_model` if it detects any invalid points (otherwise it's `None`). So if this attribute is not `None` then the points are all valid. But whether the points are all valid depend on the other objectives and the `enforce_all_objectives_valid` flag, so we have to do it in this sort of roundabout way.